crates/types/src/traits/state.rs

```diff
@@ -218,3 +219,18 @@ pub mod dummy {
         }
     }
 }
+
+/// A serialized consensus state for proof generation
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize, Default)]
+pub struct HotShotStateForProver<F: PrimeField> {
```

Comment:
this should simply be `HotShotState`? is there a conflicting struct? 

also `ForProver` is unclear in general context.

Reply:
maybe `HotShotStateDigest` or `HotshotLightClientState` ?

Reply:
changed to `LightClientState` in ccb06e9